src/main/java/edu/wpi/grip/core/Connection.java

```diff
@@ -29,6 +30,10 @@ public Connection(EventBus eventBus, Socket<? extends T> outputSocket, Socket<T>
         checkNotNull(inputSocket);
         checkNotNull(outputSocket);
         checkNotNull(eventBus);
+        checkArgument(!Socket.Direction.INPUT.equals(outputSocket.getDirection()),
+                "outputSocket cannot be an input socket");
+        checkArgument(!Socket.Direction.OUTPUT.equals(inputSocket.getDirection()),
+                "inputSocket cannot be an output socket");
```

Comment:
> A fundamental precept of OO systems is that an object should not expose any of its implementation details. This way, you can change the implementation without changing the code that uses the object. It follows then that in OO systems you should avoid getter and setter functions since they mostly provide access to implementation details.
> -- Allen Holub [Why getter and setter methods are evil](http://www.javaworld.com/article/2073723/core-java/why-getter-and-setter-methods-are-evil.html)